jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/StandardDescriptorProcessor.java

```diff
@@ -1054,34 +1063,41 @@ public ServletMapping addServletMapping(String servletName, XmlParser.Node node,
                     {
                         //The same path has been mapped multiple times, either to a different servlet or the same servlet.
                         //If its a different servlet, this is only valid to do if the old mapping was from a default descriptor.
```

Comment:
If it is valid to change a mapping from a default descriptor... and we allow embedded code to be used instead of a default descriptor, then why isn't it valid to do if the old mapping was from embedded?
I.e. If we define a servlet in embedded it is seen as a default, but a mapping appears to be handled differently?

Not sure exactly what should be done here... but either way probably need to comment to exact reasoning.

Reply:
If we allow descriptors to override embedded, then a user can never change a mapping that came from the default descriptor. The principle of least surprise says to me that if I define a servlet and mapping in embedded, I would expect those to be honoured, and not obscured by something from the defaults descriptor, which is mostly hidden from users buried inside `jetty-webapp` jar.

Reply:
@agree I think we agree on the action, but disagree on the description. If either a servlet or a mapping is defined by embedded code, then it should not be overridden by the default descriptor, but could be overridden by a normal descriptor.

Said another way, the default for either Servlet or Mapping comes from either embedded or if not set there, then the webdefaults xml.   Either default can be overridden by a normal descriptor.
So what I'd like to see is the same logic here as is done for a servlet.   If the mapping is already defined and we are the default descriptor and the mapping came from embedded, then skip the current mapping.

Reply:
Does the latest commit make this any clearer?